### ISSUE TITLE
MATLAB & Objective-C confused

### ISSUE THREAD

**Issue by mcg1969** [sourcerer-io/sourcerer-app#339]
I haven't written a lick of code in Objective-C, but Sourcerer sure thinks I do :-) I can understand why, but it would be great if there's a way to do a more advanced syntax analysis to differentiate between the two languages with unfortunately identical file suffixes.

**Comment by asurkov** [comment]
@mcg1969 could you please provide an example of your matlab code which is recognized as objective-c. I believe our samples [1] don't cover your case.

[1] https://github.com/sourcerer-io/sourcerer-app/tree/develop/src/test/resources/samples/Matlab

**Comment by mcg1969** [comment]
Feel free to take anything from either of these:
- https://github.com/cvxr/CVX
- https://github.com/cvxr/TFOCS
Again, I've written _no_ Objective-C, so all of the 681 commits and 110775 LOC being reported [on my profile](
https://sourcerer.io/mcg1969) are actually MATLAB.

**Comment by asurkov** [comment]
I imported all samples above, and no one of '.m' files was recognized as 'objective-c' (there's two of them recognized as mathematica, but it is a different story). @yaronskaya any ideas, where where it gets broken then?

**Comment by asurkov** [comment]
here's matlab samples pull request - https://github.com/sourcerer-io/sourcerer-app/pull/342

**Comment by sergey48k** [comment]
@asurkov if this is merged and deployed, could you close the issue?

**Comment by mcg1969** [comment]
All I know is that my profile still claims I'm a prolific Objective-C developer.
https://sourcerer.io/mcg1969

**Comment by asurkov** [comment]
@sergey48k right, PR is merged, but the issue is not fixed yet. Let me follow up with @yaronskaya on this.

**Comment by anatolystansler** [comment]
Hi @mcg1969!

Your profile was updated. It's a proper Matlab developer profile now :)

Feel free to reopen the issue if you have any other problems.

**Comment by mcg1969** [comment]
Thank you for working on this! Unfortunately, something still seems amiss. 

For one thing, my pie chart has a nice big section for "Wolfram". I've never programmed in that language, ever.

Secondly, my most "important" repository, subjectively, is CVX; and for some reason, in the "Repository Facts" section, it claims that it's HTML (it's Matlab, though there is some auto-generated HTML documentation) and that it use "cookie" technology, which as far as I know, it doesn't. Again, it's fair to detect that there is HTML in there; but that seems like a reasonable pattern to expect HTML documentation (and, in my case, ReStructured Text as well) inside a code repo.

So it still seems like the language detection logic isn't quite there yet.

**Comment by mcg1969** [comment]
@anatolystansler for some reason I can't reopen the issue; your permissions must be set to disallow that.

**Comment by asurkov** [comment]
@mcg1969 Wolfram is used as a fallback language, when content of *.m or *.nb file extensions is not recognized. Do you have repos other than CVX and TFOCS, which have *.m or *.nb files?

**Comment by mcg1969** [comment]
I'm afraid I do not. But I'm quite surprised that there isn't enough distinctive MATLAB in either of these modules to properly alert you!

**Comment by mcg1969** [comment]
Actually, I've contributed to the repos in this org as well, but less: https://github.com/sqlp
